encoding.bs

```diff
@@ -2431,8 +2688,131 @@ consumers of content generated with <a>GBK</a>'s <a for=/>encoder</a>.
   <p class=note><a>Index gb18030</a> maps 0xA3 0xA0 to U+3000 rather than U+E5E5 for
   compatibility with deployed content. Therefore it cannot roundtrip.
 
- <li><p>If <a>is GBK</a> is true and <var>code point</var> is
- U+20AC, return byte 0x80.
+ <li><p>If <a for="gb18030 encoder">is GBK</a> is true and <var>code point</var> is U+20AC, then
+ return byte 0x80.
+
+ <li>
+  <p>If <a for="gb18030 encoder">is GBK</a> is false and there is a row in the table below whose
+  first column is <var>code point</var>, then return the two bytes on the same row listed in the
+  second column:
```

Comment:
As long as we layer this on top of the existing gb18030 index, we can remove all the U+EXXX PUA entries below as they are already in that table and map to the correct bytes.

An implementation that updates that table would need these U+EXXX mappings though, as evidenced by WebKit. (Which I assume has a completely separate implementation for GBK.)

With those U+EXXX mappings removed however it probably makes the most sense to create a small "gb18030 2022 index" which can be reused across the decoder and encoder as a special case when GBK is not in use. Bit more work, but overall would present the data more neatly.

Reply:
WebKit (and presumably Chromium) having updated the table (as evidenced by https://github.com/vitorroriz/WebKit/commit/b7d4f07a3aac04e70f1d91d09be57e4dd2ded612) means GBK is impacted at the moment as well. In particular for the symmetrical mappings at the top of https://github.com/whatwg/encoding/issues/312#issuecomment-2354764499. What do we want to do for GBK? Perhaps it's okay if that gets slightly less PUA as well? @hsivonen what do you think?

Reply:
GBK-1995 never been an offical standard, although it asign some characters in U+E8xx for GB+FExx, and 52 Chinese character has been replaced by GB18030-2000 for unicode extension A, remained GB code GB+FExx unchanged, ex. GB+FE9F【䶮】 mapped to U+4DAE instead of U+E863.   For information interchange, we should use official GB18030-2022 mapping table enven in GBK quotation, in order to drop these duplicate unicode code to those same GB 2 bytes character.
Windows CP 936 method is wrong way to reach the GB18030 standard, i.e. remained U+E8xx characters in ttf font file (Source Han Sans & iOS never done these)，convert program using these PUA characters and assign 0x3F to 4 bytes GB18030 characters. 
On the other hand, full BMP PUA code range is U+E000~U+F8FF, and SMP U+10000~U+10FFFF mapping to GB18030 is a GB+91308130~ only, no mapping table need for programming and future GB18030 amendments.  
See detail in https://zhuanlan.zhihu.com/p/661610604 for WAHTWG GB18030 convert program(in Chinese).